test/test_torch.py

```diff
@@ -2353,17 +2353,25 @@ def test_cat_scalars(self):
                                      'zero-dimensional.*cannot be concatenated'):
             torch.cat([x, y])
 
-    def test_cat_empty(self):
+    @staticmethod
+    def _test_cat_empty(self, use_cuda=False):
         # FIXME: this is legacy behavior and should be removed
         # when we support empty tensors with arbitrary sizes
-        x = torch.randn(4, 3, 32, 32)
-        empty = torch.randn(0)
+        if use_cuda:
+            dtype = torch.cuda.float32
+        else:
+            dtype = torch.float32
+
+        x = torch.randn((4, 3, 32, 32), dtype=dtype)
+        empty = torch.randn((0,), dtype=dtype)
```

Comment:
Isn't this only checking the case that some but not all of the inputs are empty?

Reply:
Yes. There's a case later (see line 2379, GitHub won't let me tag it) that tests when all the inputs are empty.